addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComLighting4Message.java

```diff
@@ -74,43 +80,56 @@ public static SubType fromByte(int input) throws RFXComUnsupportedValueException
     }
 
     public enum Commands {
-        UNDEFINED_0(0),
-        ON(1),
-        UNDEFINED_2(2),
-        UNDEFINED_3(3),
-        OFF(4);
+        OFF_0(0, false),
+        ON_1(1, true),
+        OFF_2(2, false),
+        ON_3(3, true),
+        OFF_4(4, false),
+        ON_5(5, true),
+        ON_7(7, true),
+        ON_9(9, true),
+        ON_12(12, true),
+        UNKNOWN(-1, false);
 
         private final int command;
+        private final boolean on;
 
-        Commands(int command) {
+        Commands(int command, boolean on) {
             this.command = command;
+            this.on = on;
         }
 
         public byte toByte() {
             return (byte) command;
         }
 
-        public static Commands fromByte(int input) throws RFXComUnsupportedValueException {
+        public boolean isOn() {
+            return on;
+        }
+
+        public static Commands fromByte(int input) {
             for (Commands c : Commands.values()) {
                 if (c.command == input) {
                     return c;
                 }
             }
-            throw new RFXComUnsupportedValueException(Commands.class, input);
+            return UNKNOWN;
         }
     }
 
-    private final static List<RFXComValueSelector> SUPPORTED_INPUT_VALUE_SELECTORS = Arrays
-            .asList(RFXComValueSelector.COMMAND, RFXComValueSelector.SIGNAL_LEVEL);
+    private final static List<RFXComValueSelector> SUPPORTED_INPUT_VALUE_SELECTORS = Arrays.asList(COMMAND, COMMAND_ID,
+            SIGNAL_LEVEL, CONTACT, MOTION);
 
-    private final static List<RFXComValueSelector> SUPPORTED_OUTPUT_VALUE_SELECTORS = Arrays
-            .asList(RFXComValueSelector.COMMAND);
+    private final static List<RFXComValueSelector> SUPPORTED_OUTPUT_VALUE_SELECTORS = Arrays.asList(COMMAND, COMMAND_ID);
 
-    public SubType subType;
-    public int sensorId;
-    public Commands command;
-    public int pulse;
-    public byte signalLevel;
+    SubType subType;
+    int sensorId;
+    int pulse;
+    Commands command;
+    int commandId;
+    Commands offCommand = Commands.OFF_4;
```

Comment:
Bad tests, which read the internal values, I'll improve the tests so everything can be private here :-)